Lib/argparse.py

```diff
@@ -1843,11 +1844,6 @@ def parse_known_args(self, args=None, namespace=None):
                     if action.default is not SUPPRESS:
                         setattr(namespace, action.dest, action.default)
 
-        # add any parser defaults that aren't present
```

Comment:
Does this block need to be moved below?  This change is breaking traitlets (and IPython) who rely on there being a default value set before the Action logic fires in `self._parse_known_args`.

Reply:
I see.

Maybe all that is needed is documentation that relying on the defaults to be there _before_ the Actions run is broken. I am pretty sure that the old behavior was not intentional and that you used to have access to mutable default values in the Actions was a weird implementation detail.

Reply:
This change on when default value is set also breaks `azure-cli`: https://github.com/Azure/azure-cli/issues/20269#issuecomment-968721183